src/docfx/publish/PublishUrlMap.cs

```diff
@@ -132,16 +132,34 @@ private PublishUrlMapItem ResolveOutputPathConflicts(IGrouping<string, PublishUr
             }
 
             // redirection file is preferred than source file
-            var redirection = conflicts.FirstOrDefault(x => x.SourcePath.Origin == FileOrigin.Redirection);
-            if (redirection != null)
+            var redirections = conflicts.Where(x => x.SourcePath.Origin == FileOrigin.Redirection).OrderBy(x => x.SourcePath.Path, PathUtility.PathComparer);
+            var nonRedirections = conflicts.Where(x => x.SourcePath.Origin != FileOrigin.Redirection)
+                .OrderBy(x => x.SourcePath.Path, PathUtility.PathComparer)
+                .Select(x => x.SourcePath.Path.Value).ToList();
```

Comment:
Change the input type to `IEnumerable`, `ToList()` can be removed

Reply:
Done.